tests/suite5/extension.test.ts

```diff
@@ -0,0 +1,24 @@
+import * as assert from "assert";
+import * as vscode from "vscode";
+import { getDocUri, waitForServerStartup } from "../helper";
+
+suite("formatting tests", () => {
+  test("Should register Smithy formatter", async () => {
+    const smithyMainUri = getDocUri("suite5/smithy/main.smithy");
+    const doc = await vscode.workspace.openTextDocument(smithyMainUri);
+    await vscode.window.showTextDocument(doc);
+    await waitForServerStartup();
+    const edits: vscode.TextEdit[] = await vscode.commands.executeCommand(
+      "vscode.executeFormatDocumentProvider",
+      smithyMainUri
+    );
+    const edit: vscode.TextEdit = edits[0];
+    const expectedRange = new vscode.Range(new vscode.Position(12, 0), new vscode.Position(12, 0));
+    const range = edit.range;
+    const newText = edit.newText;
+
+    assert.equal(edits.length, 1);
```

Comment:
Is this going to be one edit per file, or is it smaller chunks than that? If the latter I'd be a bit worried about saying exactly one in case things change.